deployer/deployer-services/src/main/java/io/hyscale/deployer/services/handler/impl/V1PodHandler.java

```diff
@@ -372,4 +386,190 @@ public boolean cleanUp() {
 	public int getWeight() {
 	    return ResourceKind.POD.getWeight();
 	}
+	
+	/**
+     * It will watch event changes in the pods from cluster based on namespace, appname and service name.
+     * @param apiClient
+     * @param appName
+     * @param serviceName
+     * @param namespace
+     */
+
+	public void watch(ApiClient apiClient, String appName, String serviceName, String namespace)
+			throws HyscaleException {
+		 OkHttpClient httpClient=getUpdatedHttpClient(apiClient.getHttpClient());
+		 apiClient.setHttpClient(httpClient);
+		 CoreV1Api api = new CoreV1Api(apiClient);
+
+		String selector = ResourceSelectorUtil.getServiceSelector(appName, serviceName);
+		PodParent podParent=getPodParent(apiClient, serviceName, namespace);
+
+		Integer replicas = null;
+		String latestPodSelector = null;
+		if (podParent.getKind().equals(ResourceKind.STATEFUL_SET.getKind())) {
+			V1StatefulSetHandler v1StatefulSetHandler = (V1StatefulSetHandler) ResourceHandlers
+					.getHandlerOf(ResourceKind.STATEFUL_SET.getKind());
+			V1StatefulSet v1StatefulSet = (V1StatefulSet) podParent.getParent();
+			latestPodSelector = v1StatefulSetHandler.getPodSelector(null, v1StatefulSet, selector);
+			replicas = v1StatefulSetHandler.getReplicas(v1StatefulSet);
+		} else if (podParent.getKind().equals(ResourceKind.DEPLOYMENT.getKind())) {
+			V1DeploymentHandler v1DeploymentHandler = (V1DeploymentHandler) ResourceHandlers
+					.getHandlerOf(ResourceKind.DEPLOYMENT.getKind());
+			
+			V1Deployment v1Deployment = (V1Deployment) podParent.getParent();
+			latestPodSelector = v1DeploymentHandler.getPodSelector(apiClient, v1Deployment, selector);
+			replicas = v1DeploymentHandler.getReplicas(v1Deployment);
+		}
+		
+		if(latestPodSelector==null) {
+			throw new HyscaleException(DeployerErrorCodes.POD_SELECTOR_NOT_FOUND);
```

Comment:
The error code takes an argument. We need to pass the pod name here or remove the {} in error code.

Reply:
here exact pod name is not available